src/radix_connect/p2p_links/link_connection_qr_data.rs

```diff
@@ -0,0 +1,173 @@
+use crate::prelude::*;
+
+json_data_convertible!(LinkConnectionQRData);
+
+/// The hash of the connection password is used to connect to the Radix Connect Signaling Server,
+/// over web sockets. The actual `ConnectionPassword` is used to encrypt all messages sent via
+/// the Signaling Server.
+#[derive(
+    Clone,
+    Copy,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    Hash,
+    derive_more::Display,
+    derive_more::Debug,
+    uniffi::Record,
+)]
+#[serde(rename_all = "camelCase")]
+#[debug(
+    "LinkConnectionQRData {{ purpose: '{purpose}', password: '{password}', public_key: '{public_key}', signature: '{signature}' }}"
+)]
+#[display("{}", self.to_obfuscated_string())]
+pub struct LinkConnectionQRData {
+    pub purpose: RadixConnectPurpose,
+    pub password: RadixConnectPassword,
+    pub public_key: Ed25519PublicKey,
+    pub signature: Exactly64Bytes,
+}
+
+impl LinkConnectionQRData {
+    pub fn new(
+        purpose: RadixConnectPurpose,
+        password: RadixConnectPassword,
+        public_key: Ed25519PublicKey,
+        signature: Exactly64Bytes,
+    ) -> Self {
+        Self {
+            purpose,
+            password,
+            public_key,
+            signature,
+        }
+    }
+}
+
+#[uniffi::export]
+pub fn new_link_connection_qr_data_sample() -> LinkConnectionQRData {
```

Comment:
ok so please split the UniFFI related functions and tests into a separate file, called `the_other_file_uniffi_fn` like we do for the rest of the files. See e.g. `sargon_build_information_uniffi_fn`